__tests__/e2e/utils/response.ts

```diff
@@ -0,0 +1,170 @@
+        return field.val.flat()
+    }
+  },
+}
+
+/**
+ * Gets the title of a field as it is displayed in a response.
+ * @param {E2eFieldMetadata} field field used to create and fill form
+ * @param {FormResponseView} formMode form response mode
+ * @returns {string} the field title displayed in the response.
+ */
+export const getResponseTitle = (
+  field: E2eFieldMetadata,
+  responseView: FormResponseView,
+): string => {
+  const isCsv =
+    responseView.mode === FormResponseMode.Encrypt && responseView.csv
+  // MyInfo fields
+  if (isMyInfoableFieldType(field) && field.myInfo) {
+    if (field.myInfo.verified) return `[MyInfo] ${field.title}`
+    return field.title
+  }
+
+  // Basic fields
+  if (field.fieldType === BasicField.Table) {
+    // Delegate the work to the table handler
+    return TABLE_HANDLER.getName(field, responseView)
+  }
+  if (field.fieldType === BasicField.Attachment) {
+    switch (responseView.mode) {
+      case FormResponseMode.Email:
+        return `[attachment] ${field.title}`
+      case FormResponseMode.Encrypt:
+        return field.title
+    }
+  }
+  if (isVerifiableFieldType(field)) {
+    if (!isCsv && field.isVerifiable && field.val)
+      return `[verified] ${field.title}`
+  }
+  return field.title
+}
+
+/**
+ * Gets answers for a field as displayed in response email or decrypted submission.
+ * Always returns an array, so caller must always loop through result.
+ * @param {E2eFieldMetadata} field field used to create and fill form
+ * @param {FormResponseView} formMode form response mode
+ * @returns {string[] | null} string array of responses, or null if the field is a non-response field and should not be represented
+ */
+export const getResponseArray = (
+  field: E2eFieldMetadata,
+  responseView: FormResponseView,
+): string[] | null => {
+  const isCsv =
+    responseView.mode === FormResponseMode.Encrypt && responseView.csv
+
+  // Deal with table first to avoid special cases later
+  if (field.fieldType === BasicField.Table) {
+    return TABLE_HANDLER.getValues(field, responseView)
+  }
+
+  switch (field.fieldType) {
+    case BasicField.Section:
+      return isCsv ? null : ['']
+    case BasicField.Image:
+    case BasicField.Statement:
+      return null
+    case BasicField.Radio:
+    case BasicField.Checkbox:
+      if (!field.val || (field.val instanceof Array && !field.val.length)) {
+        return ['']
+      }
+      return [
+        (field.val instanceof Array ? field.val : [field.val])
+          .map((selected) => {
+            return field.fieldOptions.includes(selected)
+              ? selected
+              : `Others: ${selected}`
+          })
+          .join(isCsv ? ';' : ', '),
+      ]
+    case BasicField.Date:
+      // Need to re-parse, because input is in dd/mm/yyyy format whereas response is in dd MMM yyyy format.
+      return [
+        field.val
+          ? format(
+              parse(field.val, DATE_INPUT_FORMAT, new Date()),
```

Comment:
Is this date supposed to be matching `952970366`?

Reply:
it doesn't have to! `field.val` will be equal to `14/03/2000` since it was formatted. This parse and re-format converts the `field.val` to the `14 Mar 2000` format. This is necessary since we display dates on the public form (in dd/mm/yyyy format) differently from the responses that admins see (in dd MMM yyyy format).

Of course, since field.val contains no information about the time, the time data from `952970366` will be lost. But it's not an issue since we never use the time information anyway

Reply:
Got it! Thanks a lot for the explanation @justynoh!